src/relay/backend/interpreter.cc

```diff
@@ -283,251 +361,301 @@ class Interpreter : public ExprFunctor<ObjectRef(const Expr& n)>,
     return MakeClosure(func);
   }
 
-  Array<Shape> ComputeDynamicShape(const Function& func, const Array<ObjectRef>& args) {
-    CCacheKey key(func, Target("llvm"));
-    auto cfunc = compiler_->LowerShapeFunc(key);
-    size_t arity = cfunc->inputs.size() + cfunc->outputs.size();
+  /*!
+   * \brief Returns the packed function implementing the TIR function bound to \p tir_fn_var.
+   *
+   * \param tir_fn_var Global var for the already lowered TIR function.
+   * \param all_tir_fn_vars Global vars for all lowered TIR functions the above
+   * may reference, plus \p tir_fn_var itself.
+   * \param target Target for which the TIR function should be compiled. For primitives this
+   * will be the interpreter's target_. However for shape functions this will be the generic
+   * 'cpu' target, since shape functions are always executed on the host cpu.
+   */
+  PackedFunc TIRToPackedFunc(const GlobalVar& tir_fn_var, const Array<GlobalVar>& all_tir_fn_vars,
```

Comment:
Is this useful outside for the interpreter?

Reply:
Almost certainly, figured as we pull on the build loose ends this will sort itself out.